### ISSUE TITLE
Wiki: Discussion

### ISSUE THREAD

**Issue by tukusejssirs** [IzzySoft/Adebar#9]
It would be nice to add how backup & restore should look like. I don’t say this is important, but I think it should be done.

**Comment by IzzySoft** [comment]
It's basically
- check the scripts
- optionally comment out stuff you don't want
- run the scripts (and approve each backup on your device)

Same for backup and restore. I could add the above to [this wiki page](https://github.com/IzzySoft/Adebar/wiki/Usage) – or did you have something more verbose in mind?

**Comment by tukusejssirs** [comment]
Well, wwhat Iwanted to say that in wiki there’s nearly nothing about backup & restore (except for _general_ usage). I think it would be a _bit_ more **user-friendly** to include more information about what files are created, how _exactly_ to restore the backup (there’s nothing about backup in [Wiki/Usage](https://github.com/IzzySoft/Adebar/wiki/Usage) – not even one of those words ‘backup‘ or ‘restore’).

But never mind, I forked you wiki and I’m gonna check all of it and then you’ll see. However, I’ll need your help with better understanding of some details (I’ll ask you in this issue #9 if you don’t mind).:)

Just give me some time. :date: xD

**Comment by IzzySoft** [comment]
Just updated that wiki page. Better this way?

**Comment by tukusejssirs** [comment]
Much better. :)

But now I just realised—are there two kinds of backup in `adebar`? One using TiBu and the other using `adb`? Or did you just combined the two (backup the apps using `adb` and then compressed and uploaded them using TiBu)?

`Make sure your device is connected to your local WiFi, and you've enabled TiBu's WebServer` (`$backup/tibu: ln 3)
And why do I need my _device(_ (supposedly my phone) be connected to the Internet? is it not possible to download the TiBu backup to my computer and than transfered (while restoring the apps) to my phone via usb cable?

**Comment by tukusejssirs** [comment]
Can I rename this issue to `Wiki discussion`? I just thought it might be good idea to talk about what to add/edit in the wiki before it would be implemented. Also it could serve as a place to make sure one (currently just me) understands correcly what he thinks he does.

Btw, it would be a good idea to either implement or at least mention the issue #7 in the wiki (e.g. in [here](https://github.com/IzzySoft/Adebar/wiki/Installation#requirements)).

**Comment by IzzySoft** [comment]
> are there two kinds of backup in adebar? One using TiBu and the other using adb?

Yes and no. The _TiBu_ part just creates a script to pull your TiBu backups via WiFi (using _TiBu's_ built-in web server) – so one could say this "backs up the backup". This part was implemented in an early stage of _Adebar_ – I wouldn't do so now if it were not already done.

> And why do I need my device( (supposedly my phone) be connected to the Internet?

Where does it say that? It says "to your **_local**_ WiFi". As I wrote, _TiBu_ has a built-in web server we need to access here – as _TiBu_ itself manages what to send, and you need to contact it therefore. Of course you can do that manually (I e.g. use _FolderSync_ for that). Guess why the _TiBu_ feature is turned off by default :)

> Can I rename this issue to Wiki discussion?

Feel free – it's your issue :innocent:

> Btw, it would be a good idea to either implement or at least mention the issue #7 in the wiki

Agreed – though I'd chose a different page for that. I will set one up later; let's give it a week, maybe someone else chimes in and adds some device(s).

**Comment by tukusejssirs** [comment]
All is clear now. Thank you. ;)

**Comment by IzzySoft** [comment]
Anytime! Is there anything else you feel the wiki is missing – or shall I rather close this issue? It can always be re-opened later when something pops up (or a new issue added).

**Comment by tukusejssirs** [comment]
At the moment, I would change nothing, because I’ve got some other duties (mainly school stuff), so I’m a bit busy. ;)

Okay, closing the issue :)

**Comment by IzzySoft** [comment]
Fully understood! Thanks again! Whenever you find time, there have been new features added (currently in the devel branch) :innocent:

**Comment by IzzySoft** [comment]
Aw yuck! Editing Wiki files with names containing spaces (here: `Tested Devices.md`) via the web interface **_renames**_ them, replacing the space by a minus (here: `Tested-Devices.md`)! Luckily, that breaks no links (the sidebar still has `[[Tested Devices]]`, and the link works; even links from other places still addressing the page as `Tested%20Devices` keep working – obviously Github treats `-` and space as similar. Good to know: so let's take care not having two "similar names" here, not sure what happens then (if we e.g. have `Tested Devices.md` **_and**_ `Tested-Devices.md`) …

That confuses the heck out of me – definitely not something one would expect. So for the future, we better avoid file names with spaces altogether (for new files at least).

**Comment by tukusejssirs** [comment]
As fas I as know, yes, it is true, that is substitutes a ` `with`-`, but not always – just at these two arbitrary files:
- [asdf jkl](https://github.com/tukusejssirs/Programy_v_C/blob/Algoritmy-a-struktury-udajov/asdf jkl),
- [asdf-jkl](https://github.com/tukusejssirs/Programy_v_C/blob/Algoritmy-a-struktury-udajov/asdf-jkl).

You can still substitute the ` `with`%20`.

Anyway, I usually tend to _not_ use spaces in file names (and upper-cased letters and any special chars, too) just because in Bash I don’t need to escape them [it’s kinda annoying to edit just copied `$PWD` (`echo -n $PWD | xclip -selection c`) to `cd` to that dir).

**Comment by IzzySoft** [comment]
Huh? `cd "$(pwd)"` does that job just fine (uh, wait: what? That would be the same as `cd .` _LOL_). But yes, it's prone to cause errors in scripts, which is why I usually avoid it as well.

**Comment by tukusejssirs** [comment]
Indeed, `cd "$(pwd)"` does the job just fine – I did know that double quotes does the job, but never associated this to paths. :laughing:

Just to explain, how exactly I use this. I have set a function in `~/.bashrc`:

```
bash ccopy(){  # command copy
echo -n $($1) | xclip -selection c
}
```

Then, when I need to copy the path with `ccopy pwd`, open a new terminal, `cd ^V` and it is there. But only now I noticed that it does not escapes spaces.
